src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -605,6 +594,9 @@ public override TResult[] ToArray()
         public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> source, int count)
         {
             if (source == null) throw Error.ArgumentNull("source");
+            if (count <= 0) return EmptyPartition<TSource>.Instance;
```

Comment:
Rect: It does affect garbage collection. That effect is a generally good one (faster collection and perhaps in an earlier generation), but it is an effect that can be observed with finalisers or weak references, so it's not a completely unobservable change.